mysql_tests/test_loader.py

```diff
@@ -0,0 +1,402 @@
+    assert u.id == user.id
+    assert u.nickname == user.nickname
+    assert isinstance(u.team, Team)
+    assert u.team.id == user.team.id
+    assert u.team.name == user.team.name
+    assert isinstance(u.team.company, Company)
+    assert u.team.company.id == user.team.company.id
+    assert u.team.company.name == user.team.company.name
+
+
+async def test_adjacency_list(user):
+    group = Team.alias()
+
+    with pytest.raises(AttributeError):
+        group.non_exist()
+
+    # noinspection PyUnusedLocal
+    def loader(row, context):
+        rv = User(id=row[User.id], nickname=row[User.nickname])
+        rv.team = Team(id=row[Team.id], name=row[Team.name])
+        rv.team.parent = Team(id=row[group.id], name=row[group.name])
+        return rv
+
+    for exp in (
+        loader,
+        User.load(team=Team.load(parent=group)),
+        User.load(team=Team.load(parent=group.load("id", "name"))),
+        User.load(team=Team.load(parent=group.load())),
+    ):
+        u = (
+            await User.outerjoin(Team)
+            .outerjoin(group, Team.parent_id == group.id)
+            .select()
+            .gino.load(exp)
+            .first()
+        )
+
+        assert isinstance(u, User)
+        assert u.id == user.id
+        assert u.nickname == user.nickname
+        assert isinstance(u.team, Team)
+        assert u.team.id == user.team.id
+        assert u.team.name == user.team.name
+        assert isinstance(u.team.parent, Team)
+        assert u.team.parent.id == user.team.parent.id
+        assert u.team.parent.name == user.team.parent.name
+
+
+async def test_alias_distinct(user):
+    group = Team.alias()
+    group_company = Company.alias()
+    t1, t2, t3 = (
+        await Team.outerjoin(Company)
+        .outerjoin(group, Team.parent_id == group.id)
+        .outerjoin(group_company, group.company_id == group_company.id)
+        .select()
+        .order_by(Team.id)
+        .gino.load(
+            Team.distinct(Team.id).load(
+                company=Company.distinct(Company.id),
+                parent=group.distinct(group.id).load(
+                    company=group_company.distinct(group_company.id)
+                ),
+            )
+        )
+        .all()
+    )
+    assert t2.parent.name == t1.name
+    assert t1.company is t2.company
+    assert t2.parent.company is t3.parent.company
+
+
+async def test_alias_loader_columns(user):
+    user_alias = User.alias()
+    base_query = user_alias.outerjoin(Team).select()
+
+    query = base_query.execution_options(loader=AliasLoader(user_alias, "id"))
+    u = await query.gino.first()
+    assert u.id is not None
+
+
+async def test_multiple_models_in_one_query(bind):
+    for _ in range(3):
+        await User.create()
+
+    ua1 = User.alias()
+    ua2 = User.alias()
+    join_query = select([ua1, ua2]).where(ua1.id < ua2.id)
+    result = await join_query.gino.load((ua1.load("id"), ua2.load("id"))).all()
+    assert len(result) == 3
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/python-gino/gino/pullRequest?prid=5587950)